src/components/eip4626/user/VaultWithdrawForm.tsx

```diff
@@ -0,0 +1,208 @@
+import {
+  Text,
+  Flex,
+  Input,
+  Button,
+  Alert,
+  AlertDescription,
+  AlertIcon,
+} from "@chakra-ui/react";
+import Icon from "@/components/Icon";
+import { TransactionResult } from "@/components/TransactionResult";
+import { EvmAddress } from "@/types/types";
+import { useWriteHederaVaultApprove } from "@/hooks/eip4626/mutations/useWriteHederaVaultApprove";
+import { useWriteHederaVaultWithdraw } from "@/hooks/eip4626/mutations/useWriteHederaVaultWithdraw";
+import { useVaultProperties } from "@/hooks/useVaultProperties";
+import { formatNumberToBigint } from "@/services/util/helpers";
+import { useFormik } from "formik";
+import { useState, useMemo } from "react";
+import { useQueryClient } from "@tanstack/react-query";
+
+type Props = {
+  vaultAddress: EvmAddress;
+};
+
+export const VaultWithdrawForm = ({ vaultAddress }: Props) => {
+  const queryClient = useQueryClient();
+  const { shareUserBalance, vaultShareAddress, vaultAssetAddress } =
+    useVaultProperties(vaultAddress);
+
+  const {
+    data: withdrawResult,
+    mutateAsync: withdraw,
+    error: withdrawError,
+  } = useWriteHederaVaultWithdraw();
+
+  const {
+    data: approveResult,
+    mutateAsync: approve,
+    error: approveError,
+  } = useWriteHederaVaultApprove();
+
+  const withdrawForm = useFormik({
+    initialValues: {
+      amount: 0,
+    },
+    onSubmit: async ({ amount }) => {
+      const amountConverted = formatNumberToBigint(amount);
+
+      //@TODO show read allowance
+      //@TODO do not trigger allowance if it is enough?
+
+      await withdraw({
+        vaultAddress: vaultAddress,
+        tokenAmount: amountConverted,
+      });
+
+      queryClient.invalidateQueries();
+    },
+  });
+
+  const approveToken = (amount: number, isDeposit = false) => {
+    const amountConverted = formatNumberToBigint(amount);
+
+    approve({
+      tokenAmount: amountConverted,
+      tokenAddress: (isDeposit
+        ? vaultAssetAddress
+        : vaultShareAddress) as EvmAddress,
+      vaultAddress,
+    });
+  };
+
+  const handleUpdateAmountWithPercent = (percentage: number | string) => {
```

Comment:
> I guess suggesting the "max" option was not ideal, sorry about that. How about switching to 100 instead, and drop string type and checks for strings accordingly to simplify code a bit, wdyt

I can do that, but on my opinion it's better to use `max` as it looks cool for user eye. We can discuss in case you want.